### PR TITLE
ZEA-2031: implement handle raw html static build

### PR DESCRIPTION
**Pull request by hackerchai**

### CODE REVIEW

**Review by pan93412 — CHANGES_REQUESTED** [review]

File: internal/static/identify.go [hunk 1]
```diff
@@ -23,11 +23,15 @@ func (i *identify) PlanType() types.PlanType {
...
 }
 
 func (i *identify) PlanMeta(options plan.NewPlannerOptions) types.PlanMeta {
 
+	if utils.HasFile(options.Source, "index.html", "public/index.html", "dist/index.html") {
+		return types.PlanMeta{"framework": "html-static"}
+	}
+
 	if utils.HasFile(options.Source, "hugo.toml", "config/_default/hugo.toml") {
 		return types.PlanMeta{"framework": "hugo"}
```
Comment (pan93412) [inline internal/static/identify.go:36]: Hugo 不會有 public/index.html 嗎？我建議把 html-static 放到最下面。

**Review by pan93412 — CHANGES_REQUESTED** [review]

**Review by pan93412 — APPROVED** [review]

File: pkg/zeaburpack/main.go [hunk 2]
```diff
@@ -207,11 +210,24 @@ func Build(opt *BuildOptions) error {
...
+	if t == types.PlanTypeStatic {
+		println("Transforming build output to serverless format ...")
+		err = static.TransformServerless(*opt.ResultImage, *opt.Path, m)
+		if err != nil {
+			println("Failed to transform static serverless: " + err.Error())
+			handleBuildFailed(err)
+			return err
+		}
+	}
+
```
Comment (yuaanlin) [inline pkg/zeaburpack/main.go:222]:
```suggestion
	if t == types.PlanTypeStatic {
		println("Transforming build output to serverless format ...")
		m["outputDir"] = "."
		err = static.TransformServerless(*opt.ResultImage, *opt.Path, m)
		if err != nil {
			println("Failed to transform static serverless: " + err.Error())
			handleBuildFailed(err)
			return err
		}
	}
```

File: internal/static/TransformServerless.go [hunk 3]
```diff
@@ -11,7 +11,7 @@ import (
 
 // TransformServerless copies the static files from output to .zeabur/output/static and creates a config.json file for SPA
 func TransformServerless(image, workdir string, meta types.PlanMeta) error {
-	err := utils.CopyFromImage(image, path.Join("/src", meta["outputDir"])+"/.", path.Join(workdir, ".zeabur/output/static"))
+	err := utils.CopyFromImage(image, path.Join("/usr/share/nginx/html/static", meta["outputDir"])+"/.", path.Join(workdir, ".zeabur/output/static"))
```
Comment (yuaanlin) [inline internal/static/TransformServerless.go:14]:
```suggestion
	err := utils.CopyFromImage(image, path.Join("/src", meta["outputDir"])+"/.", path.Join(workdir, ".zeabur/output/static"))
```

**Review by yuaanlin — CHANGES_REQUESTED** [review]
The `static.TransformServerless` function is very simple, it copy the static files into `.zeabur/output/static` directory.

So when we handle static provider, it is just the same way as we handle static files of Node.js provider. The only different is `outputDir` is `./` (all source files).

**Comment by pan93412** [comment]
你又提交到binary了:(

**Pull request closed without merging**